Arbres/classe_noeud.py

```diff
@@ -2,26 +2,53 @@ class Noeud:
     # attribut de classe
     arbre_vide = None
 
+    """Constructeur d'un noeud d'un arbre binaire"""
+
     def __init__(self, v, g, d):
         self._valeur = v
         self._gauche = g
         self._droit = d
-    
+
+    # selecteurs
     def valeur(self):
         return self._valeur
-    
+
     def gauche(self):
         return self._gauche
-    
+
     def droit(self):
         return self._droit
-    
+
+    # methode de classe
     def est_vide(arbre):
         return arbre is Noeud.arbre_vide
 
-A_g = Noeud("D",None,None)
-A_1 = Noeud("A",Noeud("B",None, None),None)
-A_2 = Noeud("A",Noeud("B",None, None),Noeud("D",None,None))
-A_3 = Noeud("A",Noeud("B",None, Noeud("C",None,None)),Noeud("D",None,None))
+    def est_feuille(arbre):
+        if arbre.gauche() == Noeud.arbre_vide and arbre.droit() == Noeud.arbre_vide:
+            return True
+        return False
```

Comment:
comme on est dans la class, il est préférable d'utiliser `self`

```suggestion
    def est_feuille(self):
        if self.gauche() == Noeud.arbre_vide and self.droit() == Noeud.arbre_vide:
            return True
        return False
```